src/app/modules/item/helpers/item-permissions-constraints.spec.ts

```diff
@@ -0,0 +1,632 @@
+        canWatch: [ 'none' ] as const,
+        canView: canViewValues,
+      }),
+      giverPermissions: combinations({
+        canWatch: canWatchValues,
+        isOwner: [ true, false ] as const,
+      }),
+    })) {
+      expect(validateCanWatch(p.receiverPermissions, p.giverPermissions)).toEqual({});
+    }
+  });
+
+  const values = [ 'result', 'answer' ] as const;
+
+  values.forEach(value => {
+    it(`should be a able to set to "${value}" `, () => {
+
+      // giver can_watch == 'answer_with_grant' && receiver can_view >= 'content'
+      for (const p of combinations({
+        receiverPermissions: combinations({
+          canWatch: [ value ],
+          canView: [ 'content', 'content_with_descendants', 'solution' ] as const,
+        }),
+        giverPermissions: combinations({
+          canWatch: [ 'answer_with_grant' ] as const,
+          isOwner: [ true, false ] as const,
+        }),
+      })) {
+        expect(validateCanWatch(p.receiverPermissions, p.giverPermissions)).toEqual({});
+      }
+
+      // receiver can_view < 'content'
+      for (const p of combinations({
+        receiverPermissions: combinations({
+          canWatch: [ value ],
+          canView: [ 'none', 'info' ] as const,
+        }),
+        giverPermissions: combinations({
+          canWatch: canWatchValues,
+          isOwner: [ true, false ] as const,
+        }),
+      })) {
+        expect(validateCanWatch(p.receiverPermissions, p.giverPermissions)).toBeDefined();
+      }
+
+      // giver can_watch !== 'answer_with_grant'
+      for (const p of combinations({
+        receiverPermissions: combinations({
+          canWatch: [ value ],
+          canView: canViewValues,
+        }),
+        giverPermissions: combinations({
+          canWatch: [ 'none', 'result', 'answer' ] as const,
+          isOwner: [ true, false ] as const,
+        }),
+      })) {
+        expect(validateCanWatch(p.receiverPermissions, p.giverPermissions)).toBeDefined();
+      }
+    });
+  });
+
+  it('should be a able to set to "answer_with_grant" ', () => {
+    for (const p of combinations({
+      receiverPermissions: combinations({
+        canWatch: [ 'answer_with_grant' ] as const,
+        canView: [ 'content', 'content_with_descendants', 'solution' ] as const,
+      }),
+      giverPermissions: combinations({
+        canWatch: canWatchValues,
+        isOwner: [ true ] as const,
+      }),
+    })) {
+      expect(validateCanWatch(p.receiverPermissions, p.giverPermissions)).toEqual({});
+    }
+
+    // receiver can_view < 'content'
+    for (const p of combinations({
+      receiverPermissions: combinations({
+        canWatch: [ 'answer_with_grant' ] as const,
+        canView: [ 'none', 'info' ] as const,
+      }),
+      giverPermissions: combinations({
+        canWatch: canWatchValues,
+        isOwner: [ true, false ] as const,
+      }),
+    })) {
+      expect(validateCanWatch(p.receiverPermissions, p.giverPermissions)).toBeDefined();
```

Comment:
Isn't it `expect(validate(…).canWatch).toBeDefined()` ?
If the validate function always returns an object (empty or not), this test will always be true

Same below

Reply:
You're right, its fixed now